client/src/menu/Settings.tsx

```diff
@@ -31,10 +33,12 @@ export default class Settings extends React.Component<SettingsProps, SettingsSta
         GAME_MANAGER.saveSettings(volume);
         console.log("Loaded settings: " + JSON.stringify(volume));
     }
-    quitToMainMenu() {
+    async quitToMainMenu() {
         GAME_MANAGER.leaveGame();
         Anchor.closeSettings();
-        GAME_MANAGER.setDisconnectedState();
+        Anchor.setContent(<LoadingScreen type="disconnect"/>)
+        await GAME_MANAGER.setDisconnectedState();
+        Anchor.setContent(<StartMenu/>)
```

Comment:
These three lines should be in side the setDisconnectedState function

Reply:
In two other places, setDisconnectedState is used but the loading screen is not set to disconnect (on purpose). Though the anchor content is always set to the start menu. But, I'd rather have things handle things this way. It just makes sense that if you program a button to bring you somewhere, you should have to specify where it should bring you. I think this is best.